t/parse_dsn.t

```diff
@@ -0,0 +1,39 @@
+use strict;
+use warnings;
+
+use Test::Deep;
+use Test::More;
+use DBD::MariaDB;
+
+plan tests => 8;
+
+my @dsn_localhost_test = qw(
+    db=test;hostname=localhost;port=3306
+    dbname=test;host=localhost;port=3306
+    database=test;host=localhost:3306
+    test:localhost:3306
+    test:localhost:3306;notarealdb:notarealhost:1234
+    notarealdb:notarealhost:1234;database=test;host=localhost;port=3306
+    database=test;host=localhost;port=3306;notarealdb:notarealhost:1234
+);
+my %attrs_localhost_test = (
+    database => 'test',
+    host     => 'localhost',
+    port     => 3306,
```

Comment:
`localhost` is local unit socket, so `port` option cannot be used together with `localhost`.